### ISSUE TITLE
Filter parameters not working for TAXIICollectionSource

### ISSUE THREAD

**Issue by johnwunder** [oasis-open/cti-python-stix2#119]
Code:
```
from stix2 import TAXIICollectionSource, Filter
from taxii2client import Collection

# establish TAXII2 Collection instance
collection = Collection("https://limo.anomali.com/api/v1/taxii2/feeds/collections/107/", user="guest", password="guest")

# supply the TAXII2 collection to TAXIICollection
tc_source = TAXIICollectionSource(collection)

# build your filter -- kinda seems broken now
f1 = Filter("type","=", "indicator")

#retrieve the STIX objects
results = tc_source.query([f1], allow_custom=True)
```

That returns:
```
HTTPError: 500 Server Error: INTERNAL SERVER ERROR for url: https://limo.anomali.com/api/v1/taxii2/feeds/collections/107/objects/?match%5Bfilters%5D=match%5Btype%5D
```

The error seems to be in generating the filter parameter...it's somehow blowing up and it's formatted incorrectly.

**Comment by chisholm** [comment]
The way stix2 is invoking taxii2-client is wrong.  Looks like stix2 is creating HTTP query parameters itself, and passing them with a "filters" keyword arg; but it should be using a separate keyword arg per parameter, e.g. `collection.get_objects(type="indicator")`. See [here](https://github.com/oasis-open/cti-python-stix2/blob/c0669d7a5f71e73bc040b1114a9649c811ea7ce4/stix2/sources/taxii.py#L212).  taxii2-client does its own conversion [here](https://github.com/oasis-open/cti-taxii-client/blob/f6ceae007849642cfc13614318dbf60419b0a93c/taxii2client/__init__.py#L293).

**Comment by gtback** [comment]
Thanks, @johnwunder. Hopefully we'll have a chance to fix this soon. I've got a lot of ideas for cleaning up the interface between python-stix2 and taxii2-client.

**Comment by mbastian1135** [comment]
Just a note, if the stix2.TAXIICollection interface is altered, we need to alert Anthony as he is writing a blog post using the examples in the docs.

**Comment by mbastian1135** [comment]
Oh wow this is bad, in addition to passing the filters in the wrong format, we arent stripping the TAXII specific filters from the full query. So when we get objects back from TAXII, it looks like we re-apply the TAXII filters to the objects through our filter code, but our filter code throws out filters that are not common properties.  So "version" and "added_after" filters will never return anything.  :( ... My fault, Ill fix

**Comment by gtback** [comment]
@johnwunder can you check with the latest master if this is fixed, now that #147 is merged?

**Comment by gtback** [comment]
Going to close this, but if it's still a problem, feel free to re-open @johnwunder
